### PR TITLE
Allocate PCI BARs where needed

### PR DESCRIPTION
**Pull request by rbradford**
- bootinfo: Support specifying a region to allocate PCI BARs from
- pci: Allocate PCI BARs from advertised range
- fdt: Support adjusting Info::pci_bar_memory()

### CODE REVIEW

**Comment by rbradford** [comment]
I've tested this with my RV64 port and it works well.

File: src/pci.rs [hunk 1]
```diff
@@ -319,6 +322,59 @@ impl PciDevice {
...
+
+        //0x24 offset is last bar
+        while current_bar_offset <= 0x24 {
+            let bar = self.bars[current_bar];
+            if bar.size != 0 && bar.address == 0 {
+                if let Some(next_address) = next_address.as_mut() {
+                    match bar.bar_type {
+                        PciBarType::IoSpace | PciBarType::Unused => {}
+                        PciBarType::MemorySpace32 => {
+                            let address = ((*next_address + bar.size - 1) / bar.size) * bar.size;
```
Comment (sboeuf) [inline src/pci.rs:340]: nit: you could write a small helper for calculating the aligned address.

**Review by sboeuf — APPROVED** [review]
LGTM

File: src/pci.rs (hunk 1 above)
Comment (rbradford) [reply src/pci.rs]: I just added it in response to your comment :-)

**Pull request merged**